ggml/src/ggml-opencl/ggml-opencl.cpp

```diff
@@ -0,0 +1,3927 @@
+// SPDX-FileCopyrightText: Copyright (c) Qualcomm Innovation Center, Inc. All rights reserved
+// SPDX-License-Identifier: MIT
```

Comment:
The "all rights reserved" seems unnecessary here, and contradictory with the MIT license.

I understand that these tags are added as a matter of policy, but I am not sure that they make much sense for a project that does not have a single copyright holder. Should a contributor modify this file in the future, would they need to add themselves to the copyright text? I am not a laywer, but I cannot imagine that this notice would take precedence over the actual history of the file, but nonetheless, it would be confusing at least.

Additionally, since at least portions of this code are based on the previous opencl backend, I don't think that it is correct to assign the copyright solely to Qualcomm here.

Reply:
Yeah, it's a standard thing that QuIC contributors typically use.
I just double checked and looks like we are (llama.cpp community) is following general linuxfoundation guidelines
https://www.linuxfoundation.org/blog/blog/copyright-notices-in-open-source-software-projects
We'll remove the explicit copyright headers from each file as the main license already covers that.